torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -854,16 +854,25 @@ def run(
             else:
                 grid_info = getattr(grid, "grid_fn_str", "")
 
+            kernel_kwargs_str = ",".join(
+                f"{k}={v}" for (k, v) in launcher.config.kwargs.items()
```

Comment:
note: this doesn't collect kwargs for user-defined triton kernels:

https://gist.github.com/davidberard98/a17b200a7f4a5cb5963512c356b9fec6

(I wanted to test this to see if anything breaks if you have backslashes or quotes in your triton kernel args. But it turns out that Inductor doesn't handle them correctly anyway)

Reply:
Yep yep, we literally have no way to instrument custom triton kernels. We can't even distinguish between CUDA and custom triton kernels.  We've been talking about approaches to get there. Maybe one solution is to upstream some capability for hooks in triton/or override the triton decorator.

Reply:
well, here I'm talking about user-defined triton kernels in PT2 models, which _do_ show up in traces (but we don't get the kwargs)

Reply:
I think we are talking about the same thing (y) let me ping offline

Reply:
Tried running https://gist.github.com/davidberard98/a17b200a7f4a5cb5963512c356b9fec6 , the launcher data structure does not have the kwargs for the custom kernel, it is hidden inside the src code

```
@triton.jit
def kernel(in_ptr, out_ptr, BLOCK_SIZE: tl.constexpr, STRING_CONSTEXPR: tl.constexpr):
    ...
    
def run_kernel(x):
    y = torch.empty_like(x)
    kernel[(1,)](x, y, x.numel(), "sin")
    return y
```